instrumentation/kafka/kafka-clients/README.md

```diff
@@ -1,6 +1,145 @@
-# Settings for the Kafka client instrumentation
+# Instrumenting Kafka clients with OpenTelemetry
 
-| System property | Type | Default | Description |
-|---|---|---|---|
-| `otel.instrumentation.kafka.experimental-span-attributes` | Boolean | `false` | Enable the capture of experimental span attributes. |
-| `otel.instrumentation.kafka.client-propagation.enabled` | Boolean | `true` | Enables remote context propagation via Kafka message headers. |
\ No newline at end of file
+In order to get tracing information out of your application using Kafka clients, there are two ways to do so:
+
+* instrumenting your application by enabling the tracing on the Kafka clients;
+* using an external agent running alongside your application to add tracing;
+
+For both of them, pick the latest `OPENTELEMETRY_RELEASE` from [here](https://mvnrepository.com/artifact/io.opentelemetry.instrumentation/opentelemetry-instrumentation-api).
+
+## Instrumenting the Kafka clients based application
+
+Instrumenting the application means enabling the tracing in the Kafka clients.
+First, you need to add the dependency to the instrumented Kafka clients.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry.instrumentation</groupId>
+    <artifactId>opentelemetry-kafka-clients-2.6</artifactId>
+    <version>OPENTELEMETRY_RELEASE</version>
+</dependency>
+```
+
+Also, depending on the exporter that you want to use for exporting tracing information, you have to add the corresponding dependency as well.
+For example, in order to use the Jaeger exporter, the dependency is the following one.
+
+```xml
+<dependency>
+    <groupId>io.opentelemetry</groupId>
+    <artifactId>opentelemetry-exporter-jaeger</artifactId>
+    <version>OPENTELEMETRY_RELEASE</version>
+</dependency>
+```
+
+### Setting up the OpenTelemetry instance
```

Comment:
I believe that we can skip this whole part about setting up SDK, either using autoconfigure or by hand -- it's already described in detail in the [opentelemetry.io docs](https://opentelemetry.io/docs/instrumentation/java/manual/#sdk-configuration), there's no need to duplicate it here.

Reply:
As a getting started guide I think that the user should find more information here without jumping too much across links.
Even if redundant it's useful for people to get up speed soon.
I think that the link you are pointing is too generic covering even stuff that people don't need in this phase.
The auto-configuration just really need a dependency and a few env vars.
The manual configuration as well, just a dependency and a few code where we moved to reference the doc (even if maybe having the code here is anyway good as well).

Reply:
If we include it here, we should probably include it in all of the library instrumentation docs. Maybe we could move all of the library instrumentation docs to the website where they have more tooling to help automate this sort of thing, @chalin @cartermp any thoughts?

Reply:
I think the decision to make is ... what we expect from this README file available for each instrumentation library.
If I am a user, a newbie on OpenTelemetry but an expert on Kafka, what I want is to be up and running quickly and then exploring how things really works, how much they are configurable and so on. That was the purpose of my PR.
If the user starts to be linked to external doc which is more general with a tons of other things in which the user doesn't have interest at the beginning, we are just slowing down the user himself to get the technology.

Reply:
ya, I totally understand. my thought is that if we do this for the kafka instrumentation docs, then we should do this for all of the instrumentation docs, and that could be a maintenance burden with all of the duplication across so many markdown files. an alternative could be to move these docs to the website docs ([opentelemetry.io](https://opentelemetry.io/docs/instrumentation/java/)), where I think there are tools to auto-include sections into other docs which could help us manage that duplication better (and we would remove those docs from this repo).

Reply:
it's totally understandable, agree. Whatever is the solution we should aim to avoid user jumping across links but following a quick start guide in just one page.

Reply:
My personal opinion is, generally:

* Instrumentation docs that show how to use that instrumentation should probably live alongside the package (unless it's utterly ubiquitous, like Rails is with Ruby) - especially if instrumentation library authorship is sufficiently distributed across repos and organizations
* Instrumentation docs should generally assume people know how to initialize their SDK to the point where they can load an instrumentation package/component, and show an example of how to load a that point in time, then link to examples of how to initialize the SDK

> Whatever is the solution we should aim to avoid user jumping across links but following a quick start guide in just one page.

I don't necessarily agree with this as a goal. Not having to jump around is generally a good thing, but I would argue that jumping from a library's docs to the canonical docs on how to initialize the context you're working in is a pretty natural flow. I can't really think of many libraries across language ecosystems that also include docs on how to create an application that you can use that library in - they just document the pieces you need to know (like if it's middleware, where to plug it into your app's middleware pipeline, for example).

Regarding this comment:

> I think that the link you are pointing is too generic covering even stuff that people don't need in this phase.

This makes me feel like the right solution is to better organize and/or simplify the Java docs for initializing your SDK. I'm very happy to help with that.

As for moving all instrumentation docs to the repo -- I don't think that's a bad idea, so long as the READMEs for each library link back to their appropriate documentation on the website. But that would be a pretty big import of docs and so I'd like to have that be a discussion for the comms SIG if that's a direction you'd like to go.

Reply:
> an alternative could be to move these docs to the website docs ([opentelemetry.io](https://opentelemetry.io/docs/instrumentation/java/)), where I think there are tools to auto-include sections into other docs which could help us manage that duplication better (and we would remove those docs from this repo).

Personally, I like this idea -- all user-facing documentation (e.g. how to use a library instrumentation) could live on the opentelemetry.io page, while this repo would contain docs for contributors, pages explaining the internal machinery of the javaagent, and other complex stuff that 99% of people won't ever need or want to see.